packages/core-data/src/queried-data/reducer.js

```diff
@@ -134,6 +136,33 @@ const queries = flowRight( [
 	);
 } );
 
+/**
+ * Reducer tracking queries state.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Next state.
+ */
+const queries = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case 'RECEIVE_ITEMS':
+			return receiveQueries( state, action );
+		case 'REMOVE_ITEMS':
+			const newState = { ...state };
+			forEach( newState, ( queryItems, key ) => {
+				if ( newState[ key ] ) {
+					newState[ key ] = filter( queryItems, ( queryId ) => {
+						return ! action.items.includes( queryId );
```

Comment:
We should build a POJO of `action.items` to make this lookup faster. It could really hurt performance if we have a UI with lots of pagination.